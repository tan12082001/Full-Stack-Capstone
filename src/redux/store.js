import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import authenticationReducer from './authentication/authenticationSlice';

const store = configureStore({
  reducer: {
    authencation: authenticationReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
