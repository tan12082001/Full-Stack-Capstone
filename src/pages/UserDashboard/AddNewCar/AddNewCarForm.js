import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { CarSchema, carInitialValues } from '../../../models/car.model';
import { TextInputField, TextAreaInputField } from '../../../components/Form/FormField';
import FormSubmitButton from '../../../components/Button/FormSubmitButton';
import FormComponent2 from '../../../components/Form/FormComponent2';
import { postNewCar } from '../../../redux/thunk';
import { USERS_DASHBOARD } from '../../../routes/routeConstants';

const AddNewCarForm = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const status = useSelector((state) => state.cars.status);
  const handleSubmit = (values) => {
    dispatch(postNewCar(values));
  };

  if (status === 'succeeded') {
    navigate(`${USERS_DASHBOARD}`);
  }

  return (
    <FormComponent2
      initialValues={carInitialValues}
      schema={CarSchema}
      onSubmit={handleSubmit}
      className="add-new-car"
    >
      <TextInputField label="Name" name="name" placeholder="Car Name" />
      <TextAreaInputField label="Description" name="description" placeholder="Car description" />
      <TextInputField label="Price/Hour" name="pricePerHr" placeholder="Price of rent/hr" />
      <TextInputField label="Seating Capacity" name="seating_capacity" placeholder="Seating Capacity of Car" />
      <TextInputField label="Minimum Rent Duration" name="rental_duration" placeholder="Minimum Reantal Duration" />
      <FormSubmitButton type="submit" className="add-new-car-submit">
        Create Car
      </FormSubmitButton>
    </FormComponent2>
  );
};

export default AddNewCarForm;
