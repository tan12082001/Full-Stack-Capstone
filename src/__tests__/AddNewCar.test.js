import '@testing-library/jest-dom';
import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import AddNewCarForm from '../pages/UserDashboard/AddNewCar/AddNewCarForm';
import store from '../redux/store';

describe('AddNewCarForm Component', () => {
  it('renders AddNewCarForm', () => {
    const component = render(
      <React.StrictMode>
        <Provider store={store}>
          <BrowserRouter>
            <AddNewCarForm />
          </BrowserRouter>
        </Provider>
      </React.StrictMode>,
    );
    expect(component).toMatchSnapshot();
  });
});
