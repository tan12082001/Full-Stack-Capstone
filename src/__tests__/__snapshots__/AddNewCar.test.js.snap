// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewCarForm Component renders AddNewCarForm 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        action="#"
        class="add-new-car css-i51pdu"
      >
        <div
          class="css-1oq5xzy"
        >
          <label
            class="css-1mcvbvm"
          >
            Name
          </label>
          <div
            class="css-z2ctse"
          >
            <input
              class=" css-4qk13k"
              name="name"
              placeholder="Car Name"
              value=""
            />
          </div>
        </div>
        <div
          class="css-1oq5xzy"
        >
          <label
            class="css-1mcvbvm"
          >
            Description
          </label>
          <div
            class="css-z2ctse"
          >
            <textarea
              class=" css-1j5iuxi"
              name="description"
              placeholder="Car description"
              rows="10"
            />
          </div>
        </div>
        <div
          class="css-1oq5xzy"
        >
          <label
            class="css-1mcvbvm"
          >
            Price/Hour
          </label>
          <div
            class="css-z2ctse"
          >
            <input
              class=" css-4qk13k"
              name="pricePerHr"
              placeholder="Price of rent/hr"
              value=""
            />
          </div>
        </div>
        <div
          class="css-1oq5xzy"
        >
          <label
            class="css-1mcvbvm"
          >
            Seating Capacity
          </label>
          <div
            class="css-z2ctse"
          >
            <input
              class=" css-4qk13k"
              name="seating_capacity"
              placeholder="Seating Capacity of Car"
              value=""
            />
          </div>
        </div>
        <div
          class="css-1oq5xzy"
        >
          <label
            class="css-1mcvbvm"
          >
            Minimum Rent Duration
          </label>
          <div
            class="css-z2ctse"
          >
            <input
              class=" css-4qk13k"
              name="rental_duration"
              placeholder="Minimum Reantal Duration"
              value=""
            />
          </div>
        </div>
        <div
          class="css-1ssprdq"
        >
          <label
            class="css-1mcvbvm"
          >
            Add Image of Car
          </label>
          <input
            class="css-iz78xn"
            name="Add Image"
            type="file"
            value=""
          />
        </div>
        <button
          class="add-new-car-submit css-180h7py"
          type="submit"
        >
          Create Car
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      action="#"
      class="add-new-car css-i51pdu"
    >
      <div
        class="css-1oq5xzy"
      >
        <label
          class="css-1mcvbvm"
        >
          Name
        </label>
        <div
          class="css-z2ctse"
        >
          <input
            class=" css-4qk13k"
            name="name"
            placeholder="Car Name"
            value=""
          />
        </div>
      </div>
      <div
        class="css-1oq5xzy"
      >
        <label
          class="css-1mcvbvm"
        >
          Description
        </label>
        <div
          class="css-z2ctse"
        >
          <textarea
            class=" css-1j5iuxi"
            name="description"
            placeholder="Car description"
            rows="10"
          />
        </div>
      </div>
      <div
        class="css-1oq5xzy"
      >
        <label
          class="css-1mcvbvm"
        >
          Price/Hour
        </label>
        <div
          class="css-z2ctse"
        >
          <input
            class=" css-4qk13k"
            name="pricePerHr"
            placeholder="Price of rent/hr"
            value=""
          />
        </div>
      </div>
      <div
        class="css-1oq5xzy"
      >
        <label
          class="css-1mcvbvm"
        >
          Seating Capacity
        </label>
        <div
          class="css-z2ctse"
        >
          <input
            class=" css-4qk13k"
            name="seating_capacity"
            placeholder="Seating Capacity of Car"
            value=""
          />
        </div>
      </div>
      <div
        class="css-1oq5xzy"
      >
        <label
          class="css-1mcvbvm"
        >
          Minimum Rent Duration
        </label>
        <div
          class="css-z2ctse"
        >
          <input
            class=" css-4qk13k"
            name="rental_duration"
            placeholder="Minimum Reantal Duration"
            value=""
          />
        </div>
      </div>
      <div
        class="css-1ssprdq"
      >
        <label
          class="css-1mcvbvm"
        >
          Add Image of Car
        </label>
        <input
          class="css-iz78xn"
          name="Add Image"
          type="file"
          value=""
        />
      </div>
      <button
        class="add-new-car-submit css-180h7py"
        type="submit"
      >
        Create Car
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
